metrics/fid.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+import torch
+from torch.nn.functional import adaptive_avg_pool2d
+from PIL import Image
+
+from pytorch_fid.inception import InceptionV3
+from pytorch_fid.fid_score import calculate_frechet_distance
+
+def get_activations(batch, model, batch_size=16, dims=2048, device='cpu'):
+    """Calculates the activations of the pool_3 layer for all images.
+    Params:
+    -- batch       :
+    -- model       : Instance of inception model
+    -- batch_size  : Batch size of images for the model to process at once.
+                     Make sure that the number of samples is a multiple of
+                     the batch size, otherwise some samples are ignored. This
+                     behavior is retained to match the original FID score
+                     implementation.
+    -- dims        : Dimensionality of features returned by Inception
+    -- device      : Device to run calculations
+    Returns:
+    -- A numpy array of dimension (num images, dims) that contains the
+       activations of the given tensor when feeding inception with the
+       query tensor.
+    """
+    model.eval()
+
+    pred_arr = np.empty((len(batch), dims))
+
+    start_idx = 0
+
+    batch = batch.to(device)
+
+    with torch.no_grad():
+        pred = model(batch)[0]
+
+    # If model output is not scalar, apply global spatial average pooling.
+    # This happens if you choose a dimensionality not equal 2048.
+    if pred.size(2) != 1 or pred.size(3) != 1:
+        pred = adaptive_avg_pool2d(pred, output_size=(1, 1))
+
+    pred = pred.squeeze(3).squeeze(2).cpu().numpy()
```

Comment:
@annikabrundyn Currently worried about this line slowing things down (from the original implementation)